File: testdata/jennies/builders/initialization_safeguards/GoBuilder/initialization_safeguards/somepanel_builder_gen.go
package initialization_safeguards

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomePanel] = (*SomePanelBuilder)(nil)

type SomePanelBuilder struct {
    internal *SomePanel
    errors cog.BuildErrors
}

func NewSomePanelBuilder() *SomePanelBuilder {
	resource := NewSomePanel()
	builder := &SomePanelBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomePanelBuilder) Build() (SomePanel, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomePanel{}, err
	}
	
	if len(builder.errors) > 0 {
	    return SomePanel{}, cog.MakeBuildErrors("initialization_safeguards.somePanel", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *SomePanelBuilder) Title(title string) *SomePanelBuilder {
    builder.internal.Title = title

    return builder
}

func (builder *SomePanelBuilder) ShowLegend(show bool) *SomePanelBuilder {
if builder.internal.Options == nil {
    builder.internal.Options = NewOptions()
}
    builder.internal.Options.Legend.Show = show

    return builder
}

File: testdata/jennies/builders/initialization_safeguards/TypescriptBuilder/src/initialization_safeguards/somepanel_builder_gen.ts
import * as cog from '../cog';
import * as initialization_safeguards from '../initialization_safeguards';

export class SomePanelBuilder implements cog.Builder<initialization_safeguards.SomePanel> {
    private readonly internal: initialization_safeguards.SomePanel;

    constructor() {
        this.internal = initialization_safeguards.defaultSomePanel();
    }

    build(): initialization_safeguards.SomePanel {
        return this.internal;
    }

    title(title: string): this {
        this.internal.title = title;
        return this;
    }

    showLegend(show: boolean): this {
        if (!this.internal.options) {
            this.internal.options = initialization_safeguards.defaultOptions();
        }
        if (!this.internal.options.legend) {
            this.internal.options.legend = {
	show: true,
};
        }
        this.internal.options.legend.show = show;
        return this;
    }
}
